ipatests/test_integration/test_acme.py

```diff
@@ -586,17 +586,18 @@ def issue_and_expire_cert(self):
         tasks.get_kdcinfo(host)
         # Note raiseonerr=False:
         # the assert is located after kdcinfo retrieval.
-        result = host.run_command(
-            "KRB5_TRACE=/dev/stdout kinit admin",
+        # run kinit command repeatedly untill sssd gets settle
+        # after date change
+        result = tasks.run_repeatedly(
+            host, "KRB5_TRACE=/dev/stdout kinit admin",
             stdin_text='{0}\n{0}\n{0}\n'.format(
                 self.clients[0].config.admin_password
-            ),
-            raiseonerr=False
+            )
         )
         # Retrieve kdc.$REALM after the password change, just in case SSSD
         # domain status flipped to online during the password change.
         tasks.get_kdcinfo(host)
-        assert result.returncode == 0
+        assert result
```

Comment:
result here would be True/False from tasks.run_repeatedly until the command succeeds. If it times out an AssertionError is raised. So if it gets this far it will always be true. I think what you want is to check the value of get_kdcinfo() is not None to show that the SSSD domain is online (though it too seems to raise an exception on failure). So maybe just a raw call to tasks.get_kdcinfo() is sufficient.

Reply:
Otherwise this looks like a very good way to determine whether SSSD is ready or not!